spec.yaml

```diff
@@ -218,6 +231,66 @@ definitions:
           (e.g. no use of special characters) providing more information
           on the cause of the error.
         example: RESOURCE_NOT_FOUND
+  Discovery:
+    type: object
+    required:
+      - enabled
+      - apiVersion
+      - endPoint
+      - resourceTypes
+    properties:
+      enabled:
+        type: boolean
+        description: Whether the OCM service is enabled at this endpoint
+      apiVersion:
+        type: string
+        description: The OCM API version this endpoint supports
+        example: 2.0alpha
+      endPoint:
+        type: string
+        description: The URI of the OCM API available at this endpoint
+        example: https://my-cloud-storage.org/ocm
+      provider:
+        type: string
+        description: A friendly branding name of this endpoint
+        example: MyCloudStorage
+      resourceTypes:
+        type: array
+        description: A list of all supported resource types with their access protocols
+        items:
+          type: object
+          properties:
+            name:
+              type: string
+              description: |
+                A supported resource type (file, folder, calendar, contact, ...)
+              example: file
+            shareTypes:
+              type: array
+              description: The supported recipient share types
+              items:
+                type: string
+                enum: ["user", "group"]
+            protocols:
+              type: object
+              description: |
+                The supported protocols to access shares at this endpoint
+              anyOf:
+                - properties:
+                    webdav:
+                      type: string
+                      description: The top-level WebDAV endpoint
+                      example: /ocm
```

Comment:
Actually, reviewing myself, a question that I'd ask to @michielbdejong is what is the purpose of such property (other than declaring that a given protocol is supported), given that shares do have their endpoint anyway and it does not seem that share endpoints are expected to be relative to this one.

Yet I see this is how it was implemented in all EFSSs.

Reply:
Interesting! I'll test whether NC & OC-10 still work if you remove this, and if so, then it's probably better to remove it from the spec.

Reply:
Good! Concerning CERNBox/OCIS, it can be removed (i.e. this can just be specified as an enum)

Reply:
Speaking with @schiessle, it turns out we may need the this prefix. To be further discussed.